sktime/forecasting/tests/test_trend.py

```diff
@@ -2,20 +2,60 @@
 """Test trend forecasters."""
 # copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
 
-__author__ = ["mloning"]
+__author__ = ["mloning", "fkiraly"]
 __all__ = ["get_expected_polynomial_coefs"]
 
 import numpy as np
 import pandas as pd
 import pytest
 
-from sktime.forecasting.trend import PolynomialTrendForecaster
+from sktime.datasets import load_airline
+from sktime.forecasting.trend import (
+    PolynomialTrendForecaster,
+    TrendForecaster,
+    _get_X_numpy_int_from_pandas,
+)
 from sktime.utils._testing.forecasting import make_forecasting_problem
 
 
+def test_get_X_numpy():
+    """Test _get_X_numpy_int_from_pandas converts to int/float as expected."""
+    y = load_airline()
+    X_idx = _get_X_numpy_int_from_pandas(y.index)
+
+    # testing pd.PeriodIndex
+    # this should be a 2D nd.nparray, with diffs being 1 (month)
+    # because month is the periodicity
+    assert isinstance(X_idx, np.ndarray)
+    assert X_idx.shape == (len(y), 1)
+    assert (np.diff(X_idx.reshape(-1)) == 1).all()
+
+    y_idx_datetime = y.index.to_timestamp()
+    X_idx_datetime = _get_X_numpy_int_from_pandas(y_idx_datetime)
+
+    # testing pd.DatetimeIndex
+    # this should be a 2D np.ndarray, with diffs being 28, 29, 30, or 31 (days)
+    # because DatetimeIndex time stamps convert to days
+    assert isinstance(X_idx_datetime, np.ndarray)
+    assert X_idx_datetime.shape == (len(y), 1)
+    intdiffs = (np.diff(X_idx_datetime.reshape(-1))).astype(int)
```

Comment:
I think it would be worth adding a test for sub-daily data (e.g., hourly) when working with `DateTimeIndex`.

Here is an example of trying to taking the `np.diff` on the `DateTimeIndex` with hourly data:
```Python
>>> df=pd.DataFrame(data=[10,5,4,2,10], index=pd.date_range(start="2000-01-01", periods=5, freq="H"))
>>> np.diff(df.index)
array([3600000000000, 3600000000000, 3600000000000, 3600000000000],
      dtype='timedelta64[ns]')
```
Now we just want to check they're all the same.

Reply:
added